shared/language/extract.go

```diff
@@ -173,26 +154,17 @@ func ExtractAddress(db *sqlx.DB, u *dt.User, s string) (*dt.Address, bool,
 // ExtractCount returns a number from a user's message, useful in situations
 // like:
 //	Ava>  How many would you like to buy?
-//	User> Order 5.
-//
-// TODO this should return an *int64 to maintain consistency with the Extract
-// API.
-func ExtractCount(s string) sql.NullInt64 {
-	n := sql.NullInt64{
-		Int64: 0,
-		Valid: false,
-	}
+//	User> Order 5
+func ExtractCount(s string) (int64, error) {
 	s = regexNum.FindString(s)
-	if len(s) == 0 {
-		return n
+	if s == "" {
```

Comment:
No need for this change. `len` is stored internally by Go as its representation of a string, so it doesn't actually need to calculate length. For consistency we'll use `len(s) == 0`